src/modules/character.rs

```diff
@@ -28,14 +32,19 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let keymap = props.get("keymap").unwrap_or(&keymap_default);
     let exit_success = exit_code == "0";
 
+    let assumed_mode = match config.root_symbol_enabled && cfg!(target_os = "linux") && is_root_user() {
```

Comment:
does this imply linux only - i'd love to see this on mac

Reply:
macOS's root user is disabled by default, `sudo` gives your **current user** root privileges, so this doesn't make much sense on macs.

https://support.apple.com/guide/directory-utility/about-the-root-user-dirub32398f1/mac

Reply:
but on macs if i `sudo bash` i get a prompt with a `#`

Reply:
Root privileges ≠root user

Reply:
Even if using the root user is discouraged in macOS, I don't think the prompt should artificially restrict displaying that. The current implementation also seems to detect root even if the user is only acting with root privileges (`sudo bash` ).

Furthermore this check  shouldn't exclude all the other BSDs. `is_root_user()` is already handling different operating systems.